### DISCUSSION TITLE
Milvus Metrics - Could not open hwmon directory

### DISCUSSION THREAD

**Discussion by durai89** [milvus-io/milvus#15885] [Q&A and General discussion]
After we enabled metrics in server_config.yaml, we are seeing loads of error messages like `ERROR][SERVER][CPUTemperature][metric_thread] Could not open hwmon directory` .

System Spec:

- Amazon EC2 c5.9xl instance, centos7
- Milvus container: milvusdb/milvus:1.1.1-cpu-d061621-330cc6

Can you please advise what's the best approach to fix this issue? `lm_sensors `module is installed but is unable to detect any sensors on this ec2 instance. Though it would be great to report this metric, we are also happy if there is a way to blacklist this specific metric alone, so that our logs don't get bombarded with this error message?

**Comment by LoveEachDay** [comment]
Please make sure the `/sys/class/hwmon` directory exists, according to [#4240](https://github.com/milvus-io/milvus/issues/4240)

If you have installed `lm_sensors`, you can run `sensors` command to check whether any sensors enabled.

**Comment by yhmo** [comment]
The source code use a hard code path to get CPU temperature from "/sys/class/hwmon/"
```
    std::string path = "/sys/class/hwmon/";
    DIR* dir = nullptr;
    dir = opendir(path.c_str());
    if (!dir) {
        LOG_SERVER_ERROR_ << "Could not open hwmon directory";
        return result;
    }
```

Maybe the EC2 instance has no such directory. Is it possible to create an empty directory?

**Comment by yhmo** [comment]
If the EC2 instance doesn't allow creating a directory under the /sys/class,  the error cannot be avoided. You have to modify the source code and compile/build a new docker image for yourself.
The steps:
1. clone milvus project into your local
`git clone git@github.com:milvus-io/milvus.git`

2. switch to v1.1 branch and checkout to the v1.1.1 commit id
`git checkout 330cc61bede475c4a7a71841d54e633586cea829`

3. remove the error log in this cpp file: /milvus/core/src/metrics/SystemInfo.cpp
```
    if (!dir) {
        // LOG_SERVER_ERROR_ << "Could not open hwmon directory";
        return result;
    }
```

4. compile milvus under the core folder
```
cd core
./build.sh -t Release
```
5. build a new docker image
```
cd /milvus/docker/deploy/cpu/ubuntu18.04
docker build -t milvusdb/your-milvus:1.1.1-cpu .
```
6. deploy the image to the EC2

**Comment by durai89** [comment]
Thanks @LoveEachDay and @yhmo . Yes, I had already tried to create an empty directory `/sys/class/hwmon` but ec2 doesn't allow it which makes sense actually. I will try today to build the image commenting out the block of code 👍
